#! /bin/sh /usr/share/dpatch/dpatch-run
## 53_usr_share_ppd_support.dpatch by  <martin.pitt@ubuntu.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Forwarded upstream: http://www.cups.org/str.php?L2371

@DPATCH@
diff -urNad cupsys-1.2.1~/scheduler/cups-driverd.c cupsys-1.2.1/scheduler/cups-driverd.c
--- cupsys-1.2.1~/scheduler/cups-driverd.c	2006-04-18 15:16:00.000000000 +0000
+++ cupsys-1.2.1/scheduler/cups-driverd.c	2006-06-18 07:13:15.000000000 +0000
@@ -302,10 +302,7 @@
     * Try opening the file...
     */
 
-    if ((datadir = getenv("CUPS_DATADIR")) == NULL)
-      datadir = CUPS_DATADIR;
-
-    snprintf(line, sizeof(line), "%s/model/%s", datadir, name);
+    snprintf(line, sizeof(line), "/usr/share/ppd/%s", name);
     if ((fp = cupsFileOpen(line, "r")) == NULL)
     {
       fprintf(stderr, "ERROR: [cups-driverd] Unable to open \"%s\" - %s\n",
@@ -459,11 +456,7 @@
 
   SortedPPDs = NumPPDs;
 
-  if ((cups_datadir = getenv("CUPS_DATADIR")) == NULL)
-    cups_datadir = CUPS_DATADIR;
-
-  snprintf(model, sizeof(model), "%s/model", cups_datadir);
-  load_ppds(model, "");
+  load_ppds("/usr/share/ppd", "");
 
  /*
   * Cull PPD files that are no longer present...
